src/scripts/sync_synonym.py

```diff
@@ -404,6 +409,9 @@ def sync_synonyms(
     # - leave only synonyms that don't exist on given Mondo IDs
     added_df = _filter_a_by_not_in_b(source_df_with_mondo_ids, mondo_df, ['mondo_id', 'synonym_join'])
     added_df = added_df[added_df[['mondo_id', 'source_id']].apply(tuple, axis=1).isin(mapping_pairs_set)]
+    if doid_added_filtration and source_name.lower() == 'doid':
+        added_df = added_df[added_df['synonym_type'] == 'http://purl.obolibrary.org/obo/mondo#GENERATED_FROM_LABEL']
```

Comment:
This is exactly how I was hoping we do it!

Reply:
Glad to hear! You make good points on the alternative setup for the flag. I won't change it for now, but I'll add a `todo`.